src/porepy/numerics/ad/equation_system.py

```diff
@@ -477,6 +477,42 @@ def create_variables(
 
         return merged_variable
 
+    def remove_variables(self, variables: VariableList) -> None:
+        """Removes variables from the system.
```

Comment:
Should the method also delete any stored values in data[pp.TIME_STEP_SOLUTIONS]? Same for iterate.

The answer is not clear to me, it depends on the use case.

Reply:
I think not. I think of it as removing variables from the ES, not deleting them entirely.

Reply:
Let us discuss.

Reply:
Conclusion: All will be fine, provided the user remembers to update any global vectors of system states etc. (e.g., the result of `equaiton_system.get_variable_state()` obtained prior to the removal of the variable.